adobe/once.hpp

```diff
@@ -29,106 +23,31 @@ to do something here.
 
 /**************************************************************************************************/
 
-#if 0
-
-/**************************************************************************************************/
-
-namespace adobe {
-
-/**************************************************************************************************/
-
-#if defined(BOOST_HAS_THREADS)
-
-/**************************************************************************************************/
-
-typedef boost::once_flag    once_flag;
-#define ADOBE_ONCE_INIT BOOST_ONCE_INIT
-
-inline void call_once(void (*func)(), adobe::once_flag& flag)
-{
-    boost::call_once(func, flag);
-}
-
-/**************************************************************************************************/
-
-#else
-
-/**************************************************************************************************/
-
-typedef bool                once_flag;
-#define ADOBE_ONCE_INIT false
-
-inline void call_once(void (*func)(), adobe::once_flag& flag)
-{
-    if (!flag)
-    {
-        (*func)();
-        flag = true;
-    }
-}
-
-/**************************************************************************************************/
-
-#endif
-
-/**************************************************************************************************/
-
-} // namespace adobe
-
-/**************************************************************************************************/
-
-#define ADOBE_ONCE_DECLARATION(signature)                                                          \
-    struct adobe_initialize_constants_##signature##_t {                                            \
-        adobe_initialize_constants_##signature##_t();                                              \
-    };
+#ifdef ADOBE_STD_THREAD_LOCAL
 
-#define ADOBE_ONCE_DEFINITION(signature, func)                                                     \
-    namespace {                                                                                    \
-    adobe::once_flag adobe_once_flag_##signature##_s = ADOBE_ONCE_INIT;                            \
+#define ADOBE_THREAD_LOCAL_STORAGE_1(type, signature, ctor_p1)                                     \
+    type& adobe_thread_local_storage_##signature##_access() {                                      \
+        thread_local type holder {ctor_p1};                                                        \
```

Comment:
I only cleaned code that was under a   `#if 0`.
Boost.Thread fallback code is still there and used when ADOBE_STD_THREAD_LOCAL is not defined. See below.